src/scripts.js

```diff
@@ -67,40 +95,96 @@ const showRecipeView = (event) => {
+function displayIngredients(event) {
+  ingredientsList.innerHTML = '';
   const recipeId = event.target.parentNode.id;
   const currentRecipe = recipeRepo.recipes.find(recipe => 'id' + recipe.id === recipeId);
   currentRecipe.ingredients.forEach(ingredient => {
     ingredientsList.innerHTML += `<li>${ingredient.quantity.amount} ${ingredient.quantity.unit} ${ingredient.name}</li>`
   })
 }
 
-const displayDirections = (event) => {
+function displayDirections(event) {
+  directionsList.innerHTML = '';
   const recipeId = event.target.parentNode.id;
   const currentRecipe = recipeRepo.recipes.find(recipe => 'id' + recipe.id === recipeId);
   currentRecipe.instructions.forEach(direction => {
     directionsList.innerHTML += `<li>${direction.instruction}</li>`
   })
 }
 
-const filterRecipes = () => {
+function filterRecipes() {
   recipeRepo.addTag(tagInput.value);
   recipeRepo.addFilter(searchBar.value);
   recipeRepo.filterRecipes();
   createCurrentRecipes();
-  hide([homePage, recipeView]);
+  hide([homePage, recipeView, cookbook]);
   show([browsePage]);
 }
 
-window.onload = createCurrentRecipes();
-allRecipesButton.addEventListener('click', displayBrowsePage);
-submitButton.addEventListener('click', filterRecipes);
+function filterFavorites() {
+  recipeRepo.clearFilters();
+  recipeRepo.currentRecipes = user.favorites;
+  createCurrentRecipes();
+  hide([homePage, recipeView, cookbook]);
+  show([browsePage]);
+  searchBar.value = '';
+  tagInput.selectedIndex = 0;
+}
 
+function toggleFavorites(event) {
+  const heartButtons = document.querySelectorAll('.fa-heart');
+  const recipeId = event.target.parentNode.parentNode.parentNode.id;
+  const thisRecipe = recipeRepo.currentRecipes.find(recipe => "id" + recipe.id === recipeId);
+  user.favorites.includes(thisRecipe) ? 
+    user.removeFromFavorites(thisRecipe) :
+    user.addToFavorites(thisRecipe);
+  heartButtons.forEach(button => {
+    button.parentNode.parentNode.parentNode.id === recipeId ?
+      button.classList.toggle('red') : null;
+  })
+}
 
-const show = elements => elements.forEach(element => element.classList.remove('hidden'));
-const hide = elements => elements.forEach(element => element.classList.add('hidden'));
+function toggleCookbook(event) {
+  const saveButtons = document.querySelectorAll('.fa-bookmark');
+  const recipeId = event.target.parentNode.parentNode.parentNode.id;
+  const thisRecipe = recipeRepo.currentRecipes.find(recipe => "id" + recipe.id === recipeId);
+  user.cookbook.includes(thisRecipe) ? 
+    user.removeFromCookbook(thisRecipe) :
+    user.addToCookbook(thisRecipe);
+  saveButtons.forEach(button => {
+    button.parentNode.parentNode.parentNode.id === recipeId ?
+      button.classList.toggle('yellow') : null;
+  })
+}
+
+function viewCookbook() {
+  createCookbook();
+  hide([homePage, recipeView, browsePage]);
+  show([cookbook]);
+  searchBar.value = '';
+  tagInput.selectedIndex = 0;
+}
+
+function createCookbook() {
```

Comment:
This logic exists elsewhere in the program. Perhaps an opportunity to refactor to increase DRY-ness.